### DISCUSSION TITLE
Annotation-Driven Schema definitions?

### DISCUSSION THREAD

**Discussion by bdeacon-rh** [graphql-java-kickstart/graphql-java-tools#524] [Feature requests]
I don't _think_ I saw anything while poking around in the code, but as I'm putting together my .graphqls files, it seems that there should be a pretty clean way for a reflection-based class scan to be able to automatically generate the schema for a class if there were a corresponding set of annotations similar in concept to jackson's `@JsonProperty` et al.

It seems so clean, that I'm guessing it's already there and I'm just too dumb to see it. If something scanned the classes and generated the right string, that could be passed to SchemaParserBuilder.schemaString() could it not?

So my question would be: Is something like that already there? If not, is that on a roadmap? If not, is that something you think the community would be interested in seeing as a contribution?

Also, I'm new to this library. It's really well done. Good job, guys! :)

**Comment by aleksandarsusnjar** [comment]
I "feel" (not sure) that there is a "kickstart" library built on top / for `graphql-java` for exactly that. Do check. Hmmm.. just checked and didn't find it.

Anyway, I did this on my own as there are a number of "gotchas" that I needed to address:

- Consistent access control filtering that is compatible with bulk prefetching (every project is a little different)
- I don't want to expose my model exactly as it is. Nor I wanted to have to create superfluous DTOs with their own copy of the state, for each type as maintaining such DTOs in the long run, with many classes and competing needs is a nightmare.

**Reply by oliemansm** [reply]
Isn't this [graphql-annotations sample](https://github.com/graphql-java-kickstart/samples/tree/master/graphql-annotations) what you're looking for? Please note that this sample is already refactored to be working with the new 12.0.0-SNAPSHOT version. The current production version still has a separate starter for it, see https://search.maven.org/artifact/com.graphql-java-kickstart/graphql-kickstart-spring-boot-starter-graphql-annotations/11.1.0/jar.

**Reply by aleksandarsusnjar** [reply]
Yep... Looks like it is... Too bad it is tied to Spring, though.

**Reply by oliemansm** [reply]
I only pointed you to the Spring Boot starter using the GraphQL Java Annotations package. Take a look at the original library which is not a Kickstart library, but a separate one: https://github.com/Enigmatis/graphql-java-annotations.
